### PR TITLE
New documentation site + missing pages

### PR DESCRIPTION
**Pull request by theperu**
As I was writing the missing pages for our documentation I realised that I simple page like the one that we used until now was really hard to navigate and it might be difficult to use for newcomers. I looked around and I think I have found a good template for our project called [Just the Docs](https://just-the-docs.com/). By using it we would have a list of all the pages on the left and also a search bar at the top, in this PR there is also the workflow needed to update it which should work once we merge it and run automatically only when we make changes to the _docs_ folder.

Apart from this, I also tried to write something for all the missing pages which are:
- Roadmap
- Coding Style
- Widget List: for now I just put in the names of the files but ideally we should have a image and code example for each of them

I didn't create one for the DB because we will make a few changes in the near future and I think it would make much more sense to do it once we have more stable structure.

If you want to checkout how the docs will look with these changes you can do it at https://theperu.github.io/sossoldi/

Let me know what you think about all of these especially the Roadmap and Coding Style pages since are quite important and we should all decide together 😁

### CODE REVIEW

File: docs/contributing/coding-style.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+The widgets subfolder should contain widgets that are specific to a particular page, whereas the ones that are shared across multiple pages should go in /lib/custom_widgets.
+
+## Use trailing comma
+
+Flutter code often involves building fairly deep tree-shaped data structures, for example in a build method. To get good automatic formatting, we recommend you adopt the optional trailing commas. The guideline for adding a trailing comma is simple: Always add a trailing comma at the end of a parameter list in functions, methods, and constructors where you care about keeping the formatting you crafted. This helps the automatic formatter to insert an appropriate amount of line breaks for Flutter-style code.
+See here.
+
+## Avoid Copy/Paste:
+
+Do not copy and paste code blocks without proper understanding. Instead of duplicating code, consider creating a shared function or method and if you do so add a line in the [Reusable Widget List](contributing/widget-list.md).
```
Comment (mene412) [inline docs/contributing/coding-style.md:52]: This link doesn't work; I believe it needs to be modified to `contributing/widget-list.html`

File: docs/contributing/PR-guide.md [hunk 2]
```diff
@@ -1,14 +1,21 @@
...
 
-This is pretty straightforward, go to this [link](https://desktop.github.com/) and download GitHub Desktop. With this it will be easier to manage and submit the changes that you will do. 
+This is pretty straightforward, go to this [link](https://desktop.github.com/) and download GitHub Desktop. With this it will be easier to manage and submit the changes that you will do.
 After the download, set up your account.
 
 ## Step 2: Clone the repository
+
 Now comeback to your repository on GitHub and click on Code -> Open with GitHub Desktop
 
 <div>
```
Comment (mene412) [inline docs/contributing/PR-guide.md:21]: Wrong link to the resource

File: docs/contributing/PR-guide.md [hunk 3]
```diff
@@ -36,14 +43,14 @@ On the top right now you should see the option to push your changes, after that
 
 ## Step 6: Create the pull request
 
-Congrats this is the last step! Now go back to the [main repository](https://github.com/RIP-Comm/sossoldi/pulls) and create a new pull request by clicking on the top right. 
+Congrats this is the last step! Now go back to the [main repository](https://github.com/RIP-Comm/sossoldi/pulls) and create a new pull request by clicking on the top right.
 
 <div>
     <img src="./assets/doc-guide/PR-screen.png" width="1000" alt="Sossoldi icon">
```
Comment (mene412) [inline docs/contributing/PR-guide.md:49]: Wrong link to the source

File: docs/doc-guide.md [hunk 4]
```diff
@@ -24,14 +30,14 @@ GitHub already offers a great guide on how to do this but basically what you nee
...
 
 Once you've finished making all the changes you can create a commit to your fork and in a few seconds the page should update with all the changes. At this point you should verify that everything looks great and that there are no issue reaching the page.
 
 ## Step 3: Create a PR
 
-If everything went to plan now you should be able to create a PR explaining the changes that you made, in order to facilitate the review process make sure to explain what you have done and add useful screenshots (if you prefer you can just share the link *{your-username}.github.io/sossoldi/*). 
+If everything went to plan now you should be able to create a PR explaining the changes that you made, in order to facilitate the review process make sure to explain what you have done and add useful screenshots (if you prefer you can just share the link *{your-username}.github.io/sossoldi/*).
 After the review process your PR will be merged and your changes should be online!
 
 *Don't know how to open a PR? Don't worry, you can read the [guide](PR-guide.md)*
```
Comment (mene412) [inline docs/doc-guide.md:43]: This link doesn't work; I believe it needs to be modified to `PR-guide.html`
```suggestion
*Don't know how to open a PR? Don't worry, you can read the [guide](PR-guide.html)*
```

Comment (theperu) [reply docs/doc-guide.md:43]: Thanks, I'm trying to fix it now. Unfortunately I have to make a new commit every time just to see if it works

File: docs/contributing/PR-guide.md [hunk 5]
```diff
@@ -1,18 +1,25 @@
+---
+title: How to contribute to the project
+layout: default
+nav_order: 3
+has_children: true
+---
 # GitHub for beginners
 
 So you want to partecipate in our project but you don't know how? Don't worry, this guide is here for you.
 In the next steps we will assume that you already completed the environment setup. If you haven't, make sure to follow *the first 3 steps* of [this guide](setup.md) first.
```
Comment (mene412) [inline docs/contributing/PR-guide.md:10]:
```suggestion
In the next steps we will assume that you already completed the environment setup. If you haven't, make sure to follow *the first 3 steps* of [this guide](setup.html) first.
```

File: docs/setup/setup-linux-evironment.md
Comment (mene412) [inline docs/setup/setup-linux-evironment.md:1]: Wrong source link at line 233 for `Screenshot_Intellij_cmdline-tools.png`

File: docs/roadmap.md [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+- Optimize performance if necessary and address any bugs or glitches.
+
+### 2. Import/Export Data
+
+- Allow users to import historical financial data from other sources.
+- Enable export functionality to share or backup their data.
+
+### 3. Google Sheets Integration
+
+- Integrate with Google Sheets to provide users with a more in-depht environment for financial planning.
```
Comment (mikev-cw) [inline docs/roadmap.md:52]: There's a typo in "depth"

**Review by mikev-cw — COMMENTED** [review]
Really nice this template!
Well done, I've just find a little typo

File: docs/doc-guide.md (hunk 4 above)
Comment (mikev-cw) [reply docs/doc-guide.md:43]: @theperu Just do a "squash and merge" when you're done to avoid merging dozen commits

**Comment by theperu** [comment]
I fixed all the stuff that you pointed out, I'll go ahead and merge this so that we can see if we need to do something also for the deploy of the new page but hopefully everything will work as expected

**Pull request merged**